daemon/algod/api/server/lib/middlewares/auth.go

```diff
@@ -96,10 +114,19 @@ func Auth(log logging.Logger, apiToken string) func(echo.HandlerFunc) echo.Handl
 				ctx.Request().URL.Path = newPath
 			}
 
+			matchingAuthIndex := -1
 			// Check the token in constant time
-			if subtle.ConstantTimeCompare(providedToken, apiTokenBytes) == 1 {
+			for i, privAuth := range authTokens {
+				compareResult := subtle.ConstantTimeCompare(providedToken, privAuth.tokenBytes)
+				// if compareResult is 1, we want to update matchingAuthIndex with i. otherwise, we want to keep it as is.
+				matchingAuthIndex = compareResult*i + matchingAuthIndex*(1-compareResult)
+			}
+
+			if matchingAuthIndex >= 0 {
 				// Token was correct, keep serving request
-				return next(ctx)
+				if authTokens[matchingAuthIndex].paths[route] {
+					return next(ctx)
+				}
 			}
 
 			return ctx.String(http.StatusUnauthorized, "Invalid API Token")
```

Comment:
To make the error format consistent, could you change this to: `return echo.NewHTTPError(http.StatusUnauthorized, "Invalid API Token")`